pex/cli/commands/lock.py

```diff
@@ -475,17 +506,34 @@ def _update(self):
             return Error("Failed to parse project requirement to update: {err}".format(err=e))
 
         lock_file_path, lock_file = self._load_lockfile()
+        network_configuration = resolver_options.create_network_configuration(self.options)
         lock_updater = LockUpdater.create(
             lock_file=lock_file,
             repos_configuration=resolver_options.create_repos_configuration(self.options),
-            network_configuration=resolver_options.create_network_configuration(self.options),
+            network_configuration=network_configuration,
             max_jobs=resolver_options.get_max_jobs_value(self.options),
         )
 
         targets = try_(self._resolve_targets(action="updating", style=lock_file.style))
+
+        with TRACER.timed("Selecting locks to update"):
+            subset_result = try_(
+                subset(
+                    targets=targets,
+                    lock=lock_file,
+                    network_configuration=network_configuration,
+                    build=lock_file.allow_builds,
+                    use_wheel=lock_file.allow_wheels,
+                    prefer_older_binary=lock_file.prefer_older_binary,
+                    transitive=lock_file.transitive,
+                )
+            )
+
         update_requests = [
-            ResolveUpdateRequest(target=target, locked_resolve=locked_resolve)
-            for target, locked_resolve in lock_file.select(targets.unique_targets())
+            ResolveUpdateRequest(
```

Comment:
And this was the flip-over from the out-dated lock selection logic for `lock update` which came along for the ride so I could nuke the outdated logic wholesale. The existing update test still pass, vetting this.